shunting-yard-common/src/main/java/com/hotels/shunting/yard/common/emitter/WrappingMessageTask.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.shunting.yard.common.emitter;
+
+import java.util.Objects;
+
+import com.hotels.shunting.yard.common.messaging.MessageTask;
+
+public class WrappingMessageTask implements MessageTask {
+
+  private final MessageTask task;
+
+  public WrappingMessageTask(MessageTask task) {
+    this.task = task;
+  }
+
+  @Override
+  public void run() {
+    try {
+      task.run();
+      AbstractMetaStoreEventListener.success();
```

Comment:
Aah, that's why they're static. I guess an alternative to this would be for run to return a success/failure return value (or throw an exception) and the caller could call success/error based on that. Then this MessageTask wouldn't be coupled to the EventListener.

Reply:
this runs on a separate thread

Reply:
OK, but it could return a Future to whatever is running it? I'm not saying we do this but the coupling and the call to a static method on an Abstract class looks weird.

Reply:
well, this class was supposed to be inside the abstract class but then I had to make things public to allow testing so I moved it to it's own file. I don't see what's weird about it, it's like when you call static methods of the logging API in you class, perhaps the name of the class?

Reply:
I can move the static methods to an utility class but then utility classes are strange. I just want to use the same logger that the listener uses

Reply:
What did you need to make public? Wouldn't default level access with a "@visibleForTesting" annotation have worked if you wanted to keep things in one class?

Reply:
Make them protected then? Easier to test also if you don't have static stuff which is hard to mock etc. Might not be that relevant here. I guess another alternative would be to pass a non-static result listener object in that you could call methods on, that would be easy to mock if necessary. I can live with what's here too for now, I just think the risk of having static methods is that anybody can call them at any time even if they are completely unrelated to the class. This is OK for standard utility functionality like logging which you want to make "globabally" accessible. The problem here is that I could in some completely unrelated code call AbstractMetaStoreEventListener.success() and cause side effects. If you're OK with this then fine.

Reply:
no code is fool proof